server/src/test/java/com/adobe/testing/s3mock/its/ListObjectIT.java

```diff
@@ -91,13 +95,11 @@ static Param param(String prefix, String delimiter) {
         param(null, "").keys(ALL_OBJECTS), //
         param("/", null), //
         param("b", null).keys("b", "b/1", "b/1/1", "b/1/2", "b/2"), //
-        param("b/", null).keys("b/1", "b/1/1", "b/1/2", "b/2"), //

```

Comment:
The expected keys and prefixes have been derived by running those tests against an actual AWS S3 bucket. I just re-checked the correctness of this particular case. The listing will contain four keys and zero prefixes.

I, too, was unsure about the exact semantics of the ListObjects-API. That's why I implemented this test case and populated it with results from "the real thing". The key to understandit it, is not to think in directory listings but simply in string prefix comparison and splitting/aggregating at separators.

Reply:
Good point, I missed the test with different delimiters (not '/' but '~' instead) when I was running my own integration test.
I will re-run these and add test cases that do not rely solely on `/` to make sure behavior matches.